forum/qa-plugin/report-reason/q2apro-flag-reasons-validation.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+class q2apro_flag_reasons_validation {
+    protected function wrapFlagReasonError($errorCode) {
+        return ['processingFlagReasonError' => $errorCode];
+    }
+
+    protected function handleReportError($errorCode) {
+        $wrappedErrorCode = $this->wrapFlagReasonError($errorCode);
+        echo json_encode($wrappedErrorCode);
+
+        $wrappedErrorCode['_reportReasonRequestJSON'] = $this->_reportReasonRequestJSON;
+        error_log(json_encode($wrappedErrorCode));
+    }
+
+    protected function outputErrorToResponseAndExit($errorCode) {
+        $this->handleReportError(errorCode);
+        exit();
+    }
+
+    protected function isValidJSON() {
+        if (json_last_error() != JSON_ERROR_NONE) {
+            $this->handleReportError('REQUEST_IS_NOT_VALID_JSON');
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isDataSet($data) {
+        if (empty($data)) {
+            $this->handleReportError('EMPTY_REQUEST');
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isValidData($props) {
+        $propsModel = [
+            'notice' => '',
+            'postId' => 0,
+            'postType' => '',
+            'questionId' => 0,
+            'reasonId' => 0,
+            'reportType' => 'addFlag'
+        ];
+        $propsKeys = array_keys($props);
+        $matchedPropsKeysNum = 0;
+        $noticeProp = 'notice';
+        $reportTypeProp = 'reportType';
+
+//            var_dump('$props: ' ,  $props);
+
+        foreach ($propsModel as $key => $value) {
+            $propValue = @$props[$key];
```

Comment:
nie da się ładniej? nie jestem fanem @

Reply:
Da się, ale użycie małpki było prostsze. ;) Zamienić to na sprawdzanie funkcją `isset` lub `empty`?

Reply:
Ja bym zmienił. Może być nawet na `??`